File: src/main/java/pl/bk/news/api/dto/ArticleDto.java
package pl.bk.news.api.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Builder
@AllArgsConstructor
@Getter
@NoArgsConstructor
public class ArticleDto
{
    private String author;
    private String title;
    private String description;
    private String date;
    private String sourceName;
    private String articleUrl;
    private String imageUrl;
}

File: client/src/app/dto/article-dto.ts
export class ArticleDto {

  author: String;
  title: String;
  description: String;
  date: String;
  sourceName: String;
  articleUrl: String;
  imageUrl: String;

  constructor(data:any) {
    this.author = data.author || "-";
    this.title = data.title || "-";
    this.description = data.description || "-";
    this.date = data.date || "-";
    this.sourceName = data.sourceName || "-";
    this.articleUrl = data.articleUrl || "-";
    this.imageUrl = data.imageUrl || "-";
  }
}
